homeassistant/components/dyson/climate.py

```diff
@@ -146,12 +121,47 @@ def target_temperature(self):
     @property
     def current_humidity(self):
         """Return the current humidity."""
-        if self._device.environmental_state:
-            if self._device.environmental_state.humidity == 0:
-                return None
+        if (
+            self._device.environmental_state
+            and self._device.environmental_state.humidity
```

Comment:
Is it valid for the humidity to be 0? Then we should check for None instead.

Reply:
I'm not sure about this. But according to the original code, it seems that humidity equaling to 0 means invalid value.

Reply:
Ok, please add a code comment to why we're not changes this at this point.

Reply:
According to the code in `libpurecool`, humidity is set to 0 when it is not available.
https://github.com/etheralm/libpurecool/blob/e797babbcb03dd6b0a9fb9b471087ff1ad1885eb/libpurecool/dyson_pure_state.py#L115